addons/sale/models/sale.py

```diff
@@ -1804,19 +1804,20 @@ def _get_sale_order_line_multiline_description_variants(self):
             return ""
 
         name = "\n"
+        lang = self.env.context['lang'] if self.env.context.get('from_website', False) else self.order_id.partner_id.lang
 
         custom_ptavs = self.product_custom_attribute_value_ids.custom_product_template_attribute_value_id
         no_variant_ptavs = self.product_no_variant_attribute_value_ids._origin
 
         # display the no_variant attributes, except those that are also
         # displayed by a custom (avoid duplicate description)
         for ptav in (no_variant_ptavs - custom_ptavs):
-            name += "\n" + ptav.with_context(lang=self.order_id.partner_id.lang).display_name
+            name += "\n" + ptav.with_context(lang=lang).display_name
 
         # Sort the values according to _order settings, because it doesn't work for virtual records in onchange
         custom_values = sorted(self.product_custom_attribute_value_ids, key=lambda r: (r.custom_product_template_attribute_value_id.id, r.id))
         # display the is_custom values
         for pacv in custom_values:
-            name += "\n" + pacv.with_context(lang=self.order_id.partner_id.lang).display_name
+            name += "\n" + pacv.with_context(lang=lang).display_name
```

Comment:
IMHO, it would be simpler to remove the context changes in this method, and make sure `get_sale_order_line_multiline_description_sale` is called directly with the correct language (in the `product` env AND the `self` env).

Reply:
Thanks for your review @Feyensv ! Will update the fix and take a look for the test